arch/arm/src/samv7/sam_lin_sock.c

```diff
@@ -0,0 +1,1503 @@
+
+  irq_detach(priv->irq);
+}
+
+/****************************************************************************
+ * Name: sam_lin_reset
+ *
+ * Description:
+ *   Put the LIN device in the non-operational, reset state
+ *
+ * Input Parameters:
+ *   priv - reference to the private LIN driver state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+static void sam_lin_reset(struct sam_lin_s *priv)
+{
+  ninfo("LIN%" PRIu8 "\n", priv->port);
+
+  /* Reset and disable receiver and transmitter */
+
+  sam_lin_putreg(priv, SAM_UART_CR_OFFSET, UART_CR_RSTRX | UART_CR_RSTTX |
+                                           UART_CR_RXDIS | UART_CR_TXDIS);
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: sam_linsockinitialize
+ *
+ * Description:
+ *   Initialize the selected LIN port as LIN socket interface
+ *
+ * Input Parameters:
+ *   Port number (for hardware that has multiple LIN interfaces)
+ *
+ * Returned Value:
+ *   OK on success; Negated errno on failure.
+ *
+ ****************************************************************************/
+
+int sam_linsockinitialize(int port)
+{
+  struct sam_lin_s *priv = NULL;
+
+  ninfo("LIN%" PRIu8 "\n", port);
+
+  /* Select the LIN device structure */
+
+  switch (port)
+    {
+#ifdef CONFIG_SAMV7_USART0_LIN_SOCKET
+      case 0:
+        priv = &g_lin0priv;
+
+        sam_usart0_enableclk();
+
+        sam_configgpio(GPIO_USART0_RXD);
+        sam_configgpio(GPIO_USART0_TXD);
+        break;
+#endif
+
+#ifdef CONFIG_SAMV7_USART1_LIN_SOCKET
+      case 1:
+        priv = &g_lin1priv;
+
+        sam_usart1_enableclk();
+
+        sam_configgpio(GPIO_USART1_RXD);
+        sam_configgpio(GPIO_USART1_TXD);
+        break;
+#endif
+
+#ifdef CONFIG_SAMV7_USART2_LIN_SOCKET
+      case 2:
+        priv = &g_lin2priv;
+
+        sam_usart2_enableclk();
+
+        sam_configgpio(GPIO_USART2_RXD);
+        sam_configgpio(GPIO_USART2_TXD);
+        break;
+#endif
+
+      default:
+        nerr("ERROR: Unsupported port %d\n", port);
+        return -ENODEV;
+    }
+
+  /* Setup device private information */
+
+  priv->dev.d_private = priv;
+
+  /* Put the interface in the down state.  This usually amounts to resetting
+   * the device and/or calling sam_lin_ifdown().
+   */
+
+  ninfo("callbacks done\n");
+
+  sam_lin_ifdown(&priv->dev);
+
+  /* Register the device with the OS so that socket IOCTLs can be performed */
+
+  return netdev_register(&priv->dev, NET_LL_CAN);
+}
+
+/****************************************************************************
+ * Name: arm_netinitialize
+ *
+ * Description:
+ *   Initialize the LIN device interfaces.  If there is more than one device
```

Comment:
Does this description about more interfaces also apply to LIN? I think there is no reason why we could not have LIN on both USART0 and USART1 for example.

Reply:
I think that description is accurate (actually I copied it from other SocketCAN drivers, but still). It ways that function may be empty if non of the USARTs are selected as LIN and that function will add all selected USARTs as LIN over SocketCAN devices.
Yes, we can have LIN on both USART0 and USART1. We can have it even on USART0, USART1 and USART2.
Please propose what should be changed